static/js/stream_list.js

```diff
@@ -195,6 +195,9 @@ export function build_stream_list(force_rerender) {
     }
 
     $parent.append(elems);
+
+    // stream section separators should be hidden when there is only one section
+    hide_single_stream_section_separator();
```

Comment:
I think a clear implementation would be compute `needs_section_subheaders`, and just have each of the checks be, for example, `if (any_dormant_streams && needs_section_subheaders)`. It'd be a lot more readable to not have multiple controls for whether these appear, in different places.

I think doing it this way may also fix your test failure as a side effect, but not sure.

Reply:
@timabbott thanks for the feedback. I have updated the code with your feedback. However, there will be section subheaders for pinned streams as well. If we want to remove section subheaders for pinned streams, it would be best to put pin icon on the side of pinned streams to differentiate between pinned and normal streams.

Reply:
I don't understand the concern about pinned streams -- I've pushed what I had in mind for the code in a commit you can squash. The algorithm we want is "Only show headings if there's more than one section", and this achieves that.

You'll likely need some test updates to verify the new behavior.